python/ray/workflow/workflow_executor.py

```diff
@@ -0,0 +1,366 @@
+            )
+            # TODO(suquark): cancel other running tasks?
+
+    async def _post_process_ready_task(
+        self,
+        task_id: StepID,
+        metadata: WorkflowExecutionMetadata,
+        output_ref: WorkflowRef,
+    ) -> None:
+        state = self._state
+        if metadata.is_output_workflow:  # The task returns a continuation
+            sub_workflow_state: WorkflowExecutionState = await output_ref.ref
+            # init the context just for "sub_workflow_state"
+            sub_workflow_state.init_context(state.task_context[task_id])
+            state.merge_state(sub_workflow_state)
+            # build up runtime dependency
+            continuation_task_id = sub_workflow_state.output_task_id
+            state.append_continuation(task_id, continuation_task_id)
+            # Migrate callbacks - all continuation callbacks are moved
+            # under the root of continuation, so when the continuation
+            # completes, all callbacks in the continuation can be triggered.
+            if continuation_task_id in self._task_done_callbacks:
+                self._task_done_callbacks[
+                    state.continuation_root[continuation_task_id]
+                ].extend(self._task_done_callbacks.pop(continuation_task_id))
+            state.construct_scheduling_plan(sub_workflow_state.output_task_id)
+        else:  # The task returns a normal object
+            target_task_id = state.continuation_root.get(task_id, task_id)
+            state.output_map[target_task_id] = output_ref
+            if state.task_options[task_id].checkpoint:
+                state.checkpoint_map[target_task_id] = WorkflowRef(task_id)
+            state.done_tasks.add(target_task_id)
+            # TODO(suquark): cleanup callbacks when a result is set?
+            if target_task_id in self._task_done_callbacks:
+                for callback in self._task_done_callbacks[target_task_id]:
+                    callback.set_result(output_ref)
+            for m in state.reference_set[target_task_id]:
+                state.pending_input_set[m].remove(target_task_id)
+                if not state.pending_input_set[m]:
+                    state.append_frontier_to_run(m)
+
+    def _broadcast_exception(self, err: Exception):
+        for _, futures in self._task_done_callbacks.items():
+            for fut in futures:
+                if not fut.done():
+                    fut.set_exception(err)
+
+    def get_task_output_async(self, task_id: Optional[StepID]) -> asyncio.Future:
+        """Get the output of a task asynchronously.
+
+        Args:
+            task_id: The ID of task the callback associates with.
+
+        Returns:
+            A callback in the form of a future that associates with the task.
+        """
+        state = self._state
+        if self._task_done_callbacks[task_id]:
+            return self._task_done_callbacks[task_id][0]
+
+        callback = asyncio.Future()
```

Comment:
```suggestion
        fut = asyncio.Future()
```

(It's not a callback)